File: docs/api/spring/com/shoptourr/api/v1/dto/user/UserDtos.java
package com.shoptourr.api.v1.dto.user;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.util.UUID;

/**
 * Profile + Settings screens.
 */
public final class UserDtos {

    private UserDtos() {}

    public enum ThemePreference {
        SYSTEM, LIGHT, DARK
    }

    public record UserDto(
            UUID id,
            String displayName,
            String email,
            String avatarUrl,
            String locale,
            /** ISO-4217 preferred display / quote currency (mock: RUB). */
            String preferredCurrency,
            ThemePreference theme,
            boolean pushNotificationsEnabled,
            Instant memberSince,
            UserStatsDto stats
    ) {}

    public record UserStatsDto(
            int tripsCount,
            int countriesCount,
            int wishlistCount
    ) {}

    public record UpdateProfileRequest(
            @NotBlank @Size(min = 2, max = 80) String displayName,
            /** Existing media asset id after avatar upload, or null to clear. */
            UUID avatarMediaId
    ) {}

    public record UserPreferencesDto(
            @NotBlank @Size(min = 2, max = 5) String locale,
            @NotBlank @Size(min = 3, max = 3) @Pattern(regexp = "[A-Z]{3}") String preferredCurrency,
            ThemePreference theme,
            boolean pushNotificationsEnabled,
            boolean darkMode
    ) {}

    public record UpdatePreferencesRequest(
            @Size(min = 2, max = 5) String locale,
            @Size(min = 3, max = 3) @Pattern(regexp = "[A-Z]{3}") String preferredCurrency,
            ThemePreference theme,
            Boolean pushNotificationsEnabled,
            Boolean darkMode
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/data/remote/dto/user/UserDtos.kt
package com.example.shoptourr.data.remote.dto.user

import kotlinx.serialization.Serializable

@Serializable
enum class ThemePreference { SYSTEM, LIGHT, DARK }

@Serializable
enum class PremiumPlan { FREE, PLUS, PRO }

@Serializable
data class UserStatsDto(
    val tripsCount: Int,
    val countriesCount: Int,
    val wishlistCount: Int,
)

@Serializable
data class UserDto(
    val id: String,
    val displayName: String,
    val email: String,
    val avatarUrl: String? = null,
    val locale: String,
    val preferredCurrency: String,
    val theme: ThemePreference,
    val pushNotificationsEnabled: Boolean,
    val memberSince: String,
    val premiumPlan: PremiumPlan = PremiumPlan.FREE,
    val stats: UserStatsDto,
)

@Serializable
data class ActivatePremiumRequest(
    val plan: PremiumPlan,
)

@Serializable
data class UpdateProfileRequest(
    val displayName: String,
    val avatarMediaId: String? = null,
)

@Serializable
data class UserPreferencesDto(
    val locale: String,
    val preferredCurrency: String,
    val theme: ThemePreference,
    val pushNotificationsEnabled: Boolean,
    val darkMode: Boolean,
)

@Serializable
data class UpdatePreferencesRequest(
    val locale: String? = null,
    val preferredCurrency: String? = null,
    val theme: ThemePreference? = null,
    val pushNotificationsEnabled: Boolean? = null,
    val darkMode: Boolean? = null,
)

@Serializable
data class FeatureFlagsDto(
    val exportPdf: Boolean = true,
    val ocrAssist: Boolean = true,
    val nativeMaps: Boolean = false,
)

@Serializable
data class ClientRemoteConfigDto(
    val minAndroidBuild: Int,
    val minIosBuild: Int,
    val softMinAndroidBuild: Int? = null,
    val softMinIosBuild: Int? = null,
    val flags: FeatureFlagsDto = FeatureFlagsDto(),
    val storeUrlAndroid: String? = null,
    val storeUrlIos: String? = null,
)
